### DISCUSSION TITLE
Question about other data field and memory spike

### DISCUSSION THREAD

**Discussion by hjlee9182** [milvus-io/milvus#31922] [Q&A and General discussion] [answered]
Hi I have a question :)

Milvus verson. 2.3.10
collection info
- 1M rows
- 768 dim
- length 9800 varchar

1. When I load a collection, a memory spike occurs. Why does that happen?
2. When I load the same collection, the size of the memory spike varies. What could be the reason?
![image](https://github.com/milvus-io/milvus/assets/46337218/1559d297-2815-46f4-8a8c-e1342f039dad)

3. It seems like the amount of memory loaded varies when a collection contains fields of different data types, such as integers, floats, and varchars, in addition to embeddings. Is this because the different data types also get loaded into memory, affecting the overall size?
4. If the third question is correct, would the sizes for each data type be as follows?
```
bool: 1 byte
int8: 1 byte
int16: 2 bytes
int32: 4 bytes
int64: 8 bytes
float : 4 bytes
double : 8 bytes
varchar: length of string * 1bytes
```

**Comment by xiaofan-luan** [comment, marked as answer]
> Hi I have a question :)
> 
> Milvus verson. 2.3.10 collection info
> 
> * 1M rows
> * 768 dim
> * length 9800 varchar
> 
> 1. When I load a collection, a memory spike occurs. Why does that happen?
> 2. When I load the same collection, the size of the memory spike varies. What could be the reason?
>    ![image](https://private-user-images.githubusercontent.com/46337218/319908927-1559d297-2815-46f4-8a8c-e1342f039dad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIzMzQxMjcsIm5iZiI6MTcxMjMzMzgyNywicGF0aCI6Ii80NjMzNzIxOC8zMTk5MDg5MjctMTU1OWQyOTctMjgxNS00NmY0LThhOGMtZTEzNDJmMDM5ZGFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA1VDE2MTcwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4OWY1ZGIxYWRmNDQzOWNkYjQ5MzE5NjVmZmUzNWZhZTU2ZTdjMGRkMDZiNmQxODA0MWUyYjZhMDIzN2NlN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TVfLOmBC0fKJNKpEABvinDp7Gpl2Io2G9H2B3PMhBuM)
> 3. It seems like the amount of memory loaded varies when a collection contains fields of different data types, such as integers, floats, and varchars, in addition to embeddings. Is this because the different data types also get loaded into memory, affecting the overall size?
> 4. If the third question is correct, would the sizes for each data type be as follows?
> 
> ```
> bool: 1 byte
> int8: 1 byte
> int16: 2 bytes
> int32: 4 bytes
> int64: 8 bytes
> float : 4 bytes
> double : 8 bytes
> varchar: length of string * 1bytes
> ```

when load happens, there are many concurrency file handling and copy, that is why it nead actual memory to load. We recommend to leave at least 2GB of memory buffer for load.

yest you can estimate the memory usage like that, this is not accurate (usually more than that with some overhead)

**Reply by hjlee9182** [reply]
Is the size of the overhead not fixed and different for each data type and data?

**Comment by yhmo** [comment]
In milvus, string is encoded by utf8, each character occupies 1 ~ 4 bytes(1 byte for ANSI and 2 ~ 4 bytes for characters in different languages). So, for English, the size of varchar is `length of string * 1bytes`. But for Chinese or Janpanese, the size of varchar is `length of string * 3bytes` or `length of string * 4bytes`
